src/fragment-components/a-overflow-menu.tsx

```diff
@@ -0,0 +1,253 @@
+				light
+				value={item.link}
+				labelText='Link'
+				onChange={(event: any) => {
+					handleStepUpdate('link', event.currentTarget.value, index);
+				}}
+			/>
+			}
+			<div style={{ display: 'flex', flexWrap: 'wrap' }}>
+				<Checkbox
+					labelText='Has link'
+					id={`hasLink-${index}`}
+					checked={item.hasLink}
+					onChange={(checked: boolean) => handleStepUpdate('hasLink', checked, index) }/>
+				<Checkbox
+					labelText='Disabled'
+					id={`disabled-${index}`}
+					checked={item.disabled}
+					onChange={(checked: boolean) => handleStepUpdate('disabled', checked, index)}/>
+				<Checkbox
+					labelText='Is delete'
+					id={`isDelete-${index}`}
+					checked={item.isDelete}
+					onChange={(checked: boolean) => handleStepUpdate('isDelete', checked, index)} />
+				<Checkbox
+					labelText='Has divider'
+					id={`hasDivider-${index}`}
+					checked={item.hasDivider}
+					onChange={(checked: boolean) => handleStepUpdate('hasDivider', checked, index)} />
+			</div>
+		</>;
+	};
+	const updateStepList = (newList: any[]) => {
+		setComponent({
+			...selectedComponent,
+			items: newList
+		});
+	};
+	return <>
+	<Checkbox
+			labelText='Flip left'
+			id='flipped'
+			checked={selectedComponent.flipped}
+			onChange={(checked: boolean) => setComponent({
+				...selectedComponent,
+				flipped: checked
+		})} />
+	<Dropdown
+			label='Placement'
+			titleText='Placement'
+			items={placementItems}
+			initialSelectedItem={placementItems.find(item => item.id === selectedComponent.placement)}
+			itemToString={(item: any) => (item ? item.text : '')}
+			onChange={(event: any) => setComponent({
+				...selectedComponent,
+				placement: event.selectedItem.id
+	})} />
+	<DraggableTileList
+			dataList={[...selectedComponent.items]}
+			setDataList={updateStepList}
+			updateItem={handleStepUpdate}
+			defaultObject={{
+				itemText: 'New Option',
+				disabled: false,
+				hasLink: false,
+				isDelete: false,
+				hasDivider: false,
+				link: ''
+			}}
+			template={template} />
+	</>;
+};
+
+export const AOverflowMenuCodeUI = ({ selectedComponent, setComponent }: any) => {
+	return <TextInput
+			value={selectedComponent.codeContext?.name}
+			labelText='Input name'
+			onChange={(event: any) => {
+				setComponent({
+					...selectedComponent,
+					codeContext: {
+						...selectedComponent.codeContext,
+						name: event.currentTarget.value
+					}
+				});
+			}}
+		/>;
+};
+
+export const AOverflowMenu = ({
+	componentObj,
+	...rest
+}: any) => {
+	return (
+		<AComponent
+		componentObj={componentObj}
+		{...rest}>
+				<OverflowMenu
+						flipped={componentObj.flipped}
+						direction={componentObj.placement}
+						className={` ${preventCheckEvent} ${componentObj.cssClasses?.map((cc: any) => cc.id).join(' ')} `}>
+							{
+							componentObj.items.map((step: any, index: number) => (
+								<OverflowMenuItem
+									className={step.className}
+									href={step.hasLink ? step.link : undefined}
```

Comment:
`hasLink` _and_ `link`?

Reply:
yes, if the user checks off the hasLink checkbox then we enable the link input and include it in the overflow menu item

Reply:
We don't have to store hasLink in the componentObj tho, do we? We'll know if it "has link" based on `link` not being undefined.
Checkbox can be a UI thing and when it's unchecked we can clear `link` value. Perhaps after a second or two delay if it's not checked again.